analyses/snv-callers/util/wrangle_functions.R

```diff
@@ -0,0 +1,260 @@
+}
+
+wxs_bed_filter <- function(maf_df, wxs_bed = NULL, bp_window = 0) {
+  # Given a MAF formatted data.frame and a BED regions data.frame; filter out
+  # any variants of the MAF df that are not within the BED regions.
+  #
+  # Args:
+  #   maf_df: maf data that has been turned into a data.frame. Can be a maf object
+  #           that is subsetted using `@data`.
+  #   wxs_bed: a data.frame that has the windows to be used for TMB calculation.
+  #            BED formatted columns with chromosome, start, end positions in
+  #            that order.
+  #   bp_window: how many base pairs away can it be from the BED region to still
+  #              be included? Default is 0 bp. This argument gets forwarded
+  #              to GenomicRanges::findOverlaps's `maxgap` argument.
+
+  # Turn the WXS bed regions into a GRanges object
+  wxs_bed_granges <- GenomicRanges::GRanges(
+    seqnames = wxs_bed$X1,
+    ranges = IRanges::IRanges(
+      start = wxs_bed$X2,
+      end = wxs_bed$X3
+    )
+  )
+
+  # Obtain a MAF data.frame of only the WXS samples
+  maf_wxs <- maf_df %>%
+    dplyr::filter(experimental_strategy == "WXS")
+
+  # Error catcher
+  if (nrow(maf_wxs) == 0) {
+    warning("No WXS samples found underneath column 'experimental_strategy'
+            double check filtering steps and data.")
+  }
+
+  # Find the overlap of the BED regions and the mutations.
+  wxs_maf_granges <- GenomicRanges::GRanges(
+    seqnames = maf_wxs$Chromosome,
+    ranges = IRanges::IRanges(
+      start = maf_wxs$Start_Position,
+      end = maf_wxs$End_Position
+    )
+  )
+
+  # Find the overlap of the BED regions and the mutations.
+  overlap <- GenomicRanges::findOverlaps(wxs_maf_granges, wxs_bed_granges,
+    maxgap = bp_window
+  )
+
+  # Calculate of ratio of variants in this BED
+  ratio <- length(overlap@from) / nrow(maf_wxs)
+
+  # What fraction of mutations are in these bed regions?
+  cat(
+    "Ratio of variants in this BED:", ratio,
+    "Ratio of variants being filtered out:", 1 - ratio
+  )
+
+  # Only keep those in the BED regions
+  maf_wxs <- maf_wxs[unique(overlap@from), ]
+
+  # Tack these back on to the WGS samples
+  filt_maf_df <- maf_df %>%
+    dplyr::filter(experimental_strategy == "WGS") %>%
+    dplyr::bind_rows(maf_wxs)
+
+  # Return this filtered matrix
+  return(filt_maf_df)
+}
+
+calculate_tmb <- function(maf_df, wgs_size, wxs_size) {
+  # Calculate Tumor Mutational Burden for each sample given their WGS or WXS
+  # sizes in bp. Ideally have filtered the WXS mutations previously using the
+  # `wxs_bed_filter` function.
+  #
+  # TMB = # variants / size of the genome or exome surveyed
+  #
+  # Args:
+  #   maf_df: maf data.frame that has been turned into a data.frame and has
+  #           WXS mutations filtered using `wxs_bed_filter` function.
+  #   wgs_size: genome size in bp to be used for WGS samples
+  #   wxs_size: genome size in bp to be used for WGS samples
+  #
+  # Make a genome size variable
+  tmb <- maf_df %>%
```

Comment:
Remind me - does the `experimental_strategy` and `short_histology` information come with the MAF files or is that part of your set up?

Reply:
Part of the set up, but because these function requires it I've added it as an option in the function that sets up these data so that is more apparent.